design-documents/elasticsearch/extended-analyzer-config.md

```diff
@@ -0,0 +1,278 @@
+
+Each direct child of `tokenizer` element must contain `type` element with the name of tokenizer to be used.
+
+Each direct child of `tokenizer` element may contain element which represent configuration parameter available for specified tokenizer type. See [ElasticSearch parameters XML representation](#ElasticSearch-parameters-XML-representation) section for more information.
+
+```xml
+<config>
+    <tokenizer>
+        <default>
+            <type>standard</type>
+        </default>
+        <jp_JP>
+            <type>kuromoji_tokenizer</type>
+            <mode xsi:type="string">extended</mode>
+            <discard_punctuation xsi:type="boolean">false</discard_punctuation>
+            <user_dictionary xsi:type="string">userdict_ja.txt</user_dictionary>
+        </jp_JP>
+    </tokenizer>
+</config>
+```
+
+### Token Filters Configuration
+Token filters are configured by `token_filters` element which must include `default` element as a container for common configuration and may include one or more elements with valid locale codes as a name that are containers for locale-specific configuration.
+
+Each direct child of `token_filters` holds configuration of filter:
+
+- standard filters that should be used by analyzer must be declared as an empty element with name equal to token filter name (e.g. `<lowercase/>`)
+- customized filters declared by an element with name equal to custom filter name, required `type` child element which contains the name of customized token and optional parameters. See [ElasticSearch parameters XML representation](#ElasticSearch-parameters-XML-representation) section for more information.
+
+```xml
+<config>
+    <toke_filters>
+        <default>
+            <lowercase /><!-- declare usage of standard filter -->
+            <my_token_filter><!-- customized token filter -->
+                <type>standard</type>
+                <max_token_length xsi:type="number">5</max_token_length>
+            </my_token_filter>
+        </defualt>
+        <en_US><!-- locale specific filters>
+            <!-- ... -->
+        </en_US>
+    </toke_filters>
+</config>
+```
+
+### Char Filters Configuration
+Char filters configuration is similar to token filters but declared by `char_filters` element
+
+```xml
+<config>
+    <char_filters>
+        <default>
+            <html_strip /><!-- declare usage of standard filter -->
+            <my_char_filter><!-- customized filter to replece "-" by "_" -->
+                <type>pattern_replace</type>
+                <pattern>(\\d+)-(?=\\d)</pattern>
+                <replacement>$1_</replacement>
+            </my_char_filter>
+        </defualt>
+        <en_US><!-- locale specific filters>
+            <!-- ... -->
+        </en_US>
+    </char_filters>
+</config>
+```
+
+### ElasticSearch parameters XML representation
+
+JSON configuration described at ElasticSearch configuration may be converted to XML configuration to specify tokenizer and filter parameters.
```

Comment:
Where this can be specified? Could you expand on this a little or point me to where it's described in case I'm missing?

Reply:
Not sure that understand the question. XML config in the proposal does not declare strict schema. Instead of that proposed straightforward rules how to represent ElasticSerach config in XML

Reply:
Do you think there could be a benefit in making this configuration strict? Will be easier for developer to write. It's also better in case Elastic change configuration format. Could you point me to JSON configuration we trying to represent in XML?

Reply:
No. We will not cover all possible ES config options with specific elements so will have to introduce general elements like `standard_filter`, `custom_filter`, `filter_param`, etc. This will make validation of XML more strong but in cost of extra verbosity and worse readability